--- a/geos.py	2018-10-17 23:02:05.000000000 -0700
+++ b/geos.py	2018-10-17 23:16:33.017257372 -0700
@@ -39,6 +39,11 @@ def load_dll(libname, fallbacks=None):
         except OSError:
             LOG.warn("Failed `CDLL(%s)`", lib)
             pass

+        base = os.path.dirname(os.path.dirname(__file__))
+        try:
+            return CDLL(os.path.join(base, 'rpathed', 'usr', 'lib', 'x86_64-linux-gnu', lib))
+        except OSError:
+            pass
     if not dll and fallbacks is not None:
         for name in fallbacks:
             try:
